### PR TITLE
[DOCS] Adds limitation item about restarting DFA job after upgrade

### PR DESCRIPTION
**Pull request by szabosteve**
## Preview

This PR adds a new item to the DFA limitations list about jobs that are restarting after an Elasticsearch upgrade.

### Overview

[DFA restart limitation](https://stack-docs_1384.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-dfa-limitations.html#dfa-restart)

### CODE REVIEW

File: docs/en/stack/ml/df-analytics/ml-dfa-limitations.asciidoc [hunk 1]
```diff
@@ -59,6 +59,21 @@ with the increased number of data points and how the quality of results scales
...
 
+[float]
+[[dfa-restart]]
+== {dfanalytics-jobs-cap} may restart after an {es} upgrade
+  
+A {dfanalytics-job} may be restarted from the beginning in the following cases:
+
+* the job is in progress during an {es} update,
+* the job resumes on a node with a higher version,
+* the results field mapping has changed in the destination index.
```
Comment (dimitris-athanasiou) [inline docs/en/stack/ml/df-analytics/ml-dfa-limitations.asciidoc:70]: It might be clearer to reword this to "the results format has changed requiring different mappings in the destination index"

**Review by dimitris-athanasiou — COMMENTED** [review]
Just a suggestion and only in case you agree :-)

File: docs/en/stack/ml/df-analytics/ml-dfa-limitations.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/en/stack/ml/df-analytics/ml-dfa-limitations.asciidoc]: @dimitris-athanasiou It is definitely clearer this way. I amended the text accordingly via d295874.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM
